modules/spring-config/src/main/java/org/mule/runtime/config/internal/DependencyGraphLifecycleObjectSorter.java

```diff
@@ -150,8 +150,8 @@ private int getDependencyGraphIndex(Object currentObject) {
 
         @Override
         public int compare(BeanWrapper o1, BeanWrapper o2) {
-          if (getLifeCycleObjectNameOrderMap().getOrDefault(o1.getName(), -1) > getLifeCycleObjectNameOrderMap()
-              .getOrDefault(o2.getName(), -1)) {
+          if (getLifeCycleObjectList().indexOf(o1.getName()) > getLifeCycleObjectList()
+              .indexOf(o2.getName())) {
```

Comment:
doing the `indexOf` on an `ArrayList` here is O(N) now, vs. the previous O(1) when having the `Map`. We should revert to the previous internal representation.